### DISCUSSION TITLE
How to pass x-api-key header from AWS API Gateway to /docs

### DISCUSSION THREAD

**Discussion by Petar-Luketina** [fastapi/fastapi#6350] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#
```

### Description

I am using an AWS API Gateway to manage access to my FastAPI app.
To access any API endpoint, an API key header is required (`'x-api-key': 'api-key-here'`)
Problematically, there is nowhere to pass headers when targeting the `/docs` endpoint in the browser.
How--or is--it possible to access `/docs` with a required API key header?

### Operating System

Linux, Windows

### FastAPI Version

0.74.1

### Python Version

Python 3.10.3

**Comment by odiseo0** [comment]
I had the same problem several days ago but right now it works fine, I didn't do anything.
Are you using aws lambda or what service?

**Comment by Petar-Luketina** [comment]
That's correct, I'm using AWS Lambda.
Is it just the `/docs` endpoint that started to not require an API key?
That behavior does seem strange if you did not change anything.
We would expect any endpoint that falls under the GET method requests would need an API key per the linked configurations:
https://pasteboard.co/bPKP76cpRuGs.png

**Comment by Petar-Luketina** [comment, marked as answer]
It seems like the best thing to do is exclude `/docs` from requiring an API key.
After that, add a SwaggerUI box for their API key that passed the value to a request header.
I'll close the question, too.

**Comment by Petar-Luketina** [comment]
@odiseo0 , thank you for your input btw!

**Comment by lepri** [comment]
Hi @Petar-Luketina ,

Did you able to exclude `/docs` from requiring an API key? If so can you share the solution?

Best,
